### PR TITLE
Change monitor enter return type

### PR DESCRIPTION
**Pull request by tajila**
Change monitor enter return type

With the previous code returns an IDATA for monitor status. A check is
performed to determine if the monitor status is less than the blocking
enter return code (currently 4). This is broken because a valid object
can be a negative value (highly likely on 32bit platforms).

This PR changes the return type from a IDATA to a UDATA. Also, it fixes
the bytecode action order.

Signed-off-by: Tobi Ajila <atobia@ca.ibm.com>

### CODE REVIEW

File: runtime/vm/BytecodeAction.hpp [hunk 1]
```diff
@@ -47,16 +47,16 @@ typedef enum {
...
-	THROW_VALUE_TYPE_ILLEGAL_MONITOR_STATE,
-#endif /* defined(J9VM_OPT_VALHALLA_VALUE_TYPES) */
 	THROW_ILLEGAL_MONITOR_STATE,
 	THROW_INCOMPATIBLE_CLASS_CHANGE,
 	THROW_WRONG_METHOD_TYPE,
 	PERFORM_DLT,
 	RUN_METHOD_INTERPRETED,
 	RUN_JNI_NATIVE,
 	RUN_METHOD_COMPILED,
+#if defined(J9VM_OPT_VALHALLA_VALUE_TYPES)
```
Comment (gacholio) [inline runtime/vm/BytecodeAction.hpp]: This still doesn't match the interpreter - the VT one is after the debug ones in the interpreter.

File: runtime/vm/BytecodeInterpreter.hpp [hunk 2]
```diff
@@ -9106,8 +9106,8 @@ done:;
 			goto jni; \
 		case RUN_METHOD_COMPILED: \
 			goto i2j; \
-		DEBUG_ACTIONS \
 		PERFORM_ACTION_VALUE_TYPE_IMSE \
```
Comment (tajila) [inline runtime/vm/BytecodeInterpreter.hpp]: @gacholio Are you referring to this? or something else?

Comment (gacholio) [reply runtime/vm/BytecodeInterpreter.hpp]: Sorry, missed the fact that you also changed this.

**Comment by gacholio** [comment]
jenkins test sanity zlinux jdk8

**Pull request merged**